### PR TITLE
Use non deprecated xcontenthelper

### PR DESCRIPTION
**Pull request by dakrone**
This moves away from one of the now-deprecated XContentHelper.createParser
methods in favor of specifying the deprecation logger at parser creation time.

Relates to #28449

Note that this doesn't move all the `createParser` calls because some of them
use the already-deprecated method that doesn't specify the XContentType.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java [hunk 1]
```diff
@@ -316,7 +317,8 @@ static Request bulk(BulkRequest bulkRequest) throws IOException {
                 BytesReference indexSource = indexRequest.source();
                 XContentType indexXContentType = indexRequest.getContentType();
 
-                try (XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY, indexSource, indexXContentType)) {
+                try (XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY,
+                    LoggingDeprecationHandler.INSTANCE, indexSource, indexXContentType)) {
```
Comment (nik9000) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java]: Can you indent this one more time? The way it is the method args look like they are part of the `try` body.

**Review by nik9000 — APPROVED** [review]
I left a note about indentation. Can you add four more spaces to all the `try` changes?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java (hunk 1 above)
Comment (dakrone) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java]: I checked but I thought we used what Intellij thought the indentation should be? (I even opened Intellij to make sure it agreed)

Comment (nik9000) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java]: I tend to disagree with IntelliJ here. It is hard to read..... Whatever works though.

**Comment by dakrone** [comment]
@elasticmachine test this again please

**Pull request merged**